test/index.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+var assert = require('assert');
+var intercept = require('intercept-stdout');
+
+var Z = require('sanctuary-type-classes');
+
+var createRunner = require('..');
+
+
+function eq(actual, expected) {
+  assert.strictEqual(arguments.length, eq.length);
+  assert.strictEqual(Z.toString(actual), Z.toString(expected));
+  assert.strictEqual(Z.equals(actual, expected), true);
+}
+
+function expectOutput(expectations) {
+  var idx = 0;
+  var unhook = intercept(function(text) {
+    text.split('\n').forEach(function(line) {
+      if(idx >= expectations.length) {
+        unhook();
+        throw new Error(
+          'More lines than expected were printed:\n\n' +
+          JSON.stringify(line)
+        );
+      }
+      if(expectations[idx].test(line) === false) {
+        unhook();
+        throw new Error(
+          'A printed line did not meet expectations:\n\n' +
+          JSON.stringify(line) + '\n\n' +
+          'It does not match expression:\n\n' +
+          expectations[idx].toString()
+        );
+      }
+      idx += 1;
+    });
+  });
+  return function() {
+    unhook();
+    if(idx < expectations.length - 1) {
+      throw new Error(
+        'Expected ' +
+        String(idx - expectations.length) +
+        ' more lines to be printed.'
+      );
+    }
+  };
+}
+
+var mocklibA = {lib: 'a'};
+var mocklibB = {lib: 'b'};
+
+test('createRunner', function() {
+  eq(typeof createRunner, 'function');
+  eq(createRunner.length, 3);
+  eq(typeof createRunner(mocklibA, mocklibB, {}), 'function');
+  eq(createRunner(mocklibA, mocklibB, {}).length, 1);
+});
+
+var runnerUniform = createRunner(mocklibA, mocklibB, {
+  mock: function(lib){ eq(lib.lib === 'a' || lib.lib === 'b', true); }
+});
+
+var runnerSplit = createRunner(mocklibA, mocklibB, {
+  mock: {
+    left: function(lib) { eq(lib.lib, 'a'); },
+    right: function(lib) { eq(lib.lib, 'b'); }
+  }
+});
+
+function runAndAssert(options, expectations) {
+  var done = expectOutput(expectations);
+  runnerUniform(options);
+  done();
+  var done = expectOutput(expectations);
+  runnerSplit(options);
+  done();
+}
+
+test('runBenchmarks', function() {
+  runAndAssert({
+    minSamples: 1,
+    colors: false,
+    significantChange: 0
+  }, [
+    /^# 1\/1: mock\r/,
+    /^┌[─┬]+┐$/,
+    /^│ suite +│ left +│ right +│ change +│ α │$/,
+    /^├[─┼]+┤$/,
+    /^│ mock +│(?: [0-9,]+ Hz ±\d+\.\d+% \(n \d+\) +│){2} [+-]\d+\.\d% +│ [ ✓✗] │$/,
+    /^└[─┴]+┘$/
```

Comment:
I wrote the tests in such a way that we can easily test the impact on the output that each change to the options has, but writing these expressions ended up costing some time, so for now we only test a single (color-less) configuration.